Products/ZenModel/FileSystem.py

```diff
@@ -110,11 +110,23 @@ class FileSystem(OSComponent):
         )
 
 
+    def getNoneRootTotalBlocks(self):
+        availBlocks = self.cacheRRDValue('availBlocks', None)
+        usedBlocks = self.usedBlocks()
+        if availBlocks and usedBlocks:
+            nonRootTotalBlocks = availBlocks + usedBlocks
+            return nonRootTotalBlocks
+
+
     def getTotalBlocks(self):
+        noneRootTotalBlocks = self.getNoneRootTotalBlocks()
+        if noneRootTotalBlocks:
+            return noneRootTotalBlocks
```

Comment:
From reading the result of your investigation of how to calculate the total block count, it seems Zenoss has to support two different algorithms.  It also seems like both algorithms have to exist behind the `getTotalBlocks` API.

The two algorithms are _sum of parts_ and _percent of total_, with _sum of parts_ being the preferred algorithm if the required parts are available.

No critique here; I'm just summarizing my understanding of what the fix is.